### PR TITLE
Issue710 replace event filtering v3

### LINKED ISSUE #710: Replace event filtering
Polygon nodes hosted by Infura don't support event filtering. 

But aquarius uses event filtering here: https://github.com/oceanprotocol/aquarius/blob/v4main/aquarius/events/events_monitor.py#L385-L393

When running aquarius on Polygon or Mumbai hosted by Infura, the following error occurs: 
```
2022-02-17 19:12:01,644 - aquarius.events.events_monitor - ERROR - Error processing event: {'code': -32004, 'message': 'method not supported eth_newFilter'}.
```

This can be overcome by running your own Polygon/Mumbai node but this option may be difficult for some users.

Proposed Solution:
Replace event filter with a loop that calls `eth_getLogs` in 1000 block chunks, similar to ocean.py: https://github.com/oceanprotocol/ocean.py/blob/v4main/ocean_lib/web3_internal/contract_base.py#L352-L413

The chunk size should be configurable. Also, the referenced ocean.py code searches backward from the most recent block. Aquarius might want to choose the opposite behavior - searching forward from the oldest block.

### PR DESCRIPTION
**Pull request by DMats**
Fixes #710 

Changes:

- Replace event filtering
- Fix bug where CI was using v4 barge instead of v3 barge.
- Add instruction to print contents of `address.json` in CI logs
- Bump flask from 2.0.2 to 2.1.2

### CODE REVIEW

**Comment by DMats** [comment]
Hmm... It looks like there's actually an error when uploading the coverage results.
https://github.com/oceanprotocol/aquarius/runs/6426422492?check_suite_focus=true#step:9:252

> time="2022-05-13T16:32:34Z" level=warning msg="Conflict when uploading: A test report for commit 0449192a36e23c7dcbcb92ecef484c765959b3f0 already exists, skipping upload"

File: tests/test_events.py [hunk 1]
```diff
@@ -280,17 +281,23 @@ def test_add_chain_id_to_chains_list(events_object):
...
 
-    assert (
-        events_object.get_event_logs(EVENT_METADATA_CREATED, 0, 10, get_logs_exception)
-        is None
-    )
-    result = events_object.get_event_logs(EVENT_METADATA_CREATED, 0, 10, get_logs_mock)
+    with patch("web3.contract.ContractEvent.getLogs") as mock:
+        mock.side_effect = Exception("Boom!")
+
+        with pytest.raises(Exception):
```
Comment (calina-c) [inline tests/test_events.py:294]: please add match="<message>" arguments to the pytest raises part

**Review by calina-c — COMMENTED** [review]
I added a small suggestion.

**Comment by calina-c** [comment]
@DMats can you run coverage locally, to make sure the new lines are covered?

File: tests/test_events.py (hunk 1 above)
Comment (DMats) [reply tests/test_events.py:294]: Done.

**Review by calina-c — APPROVED** [review]

**Comment by DMats** [comment]
@calina-c helpled me get the provider tests running on my local machine.  The fix was to run elasticsearch in a separate terminal because the tests weren't able to connect to elasticsearch running inside barge.

Here's the coverage report with covered files omitted and missing lines enumerated.  As you can see, the lines changed by this PR are not missing from test coverage.

```
(venv) david@david-Inspiron-5502:~/projects/aquarius$ coverage report --skip-covered -m
Name                                Stmts   Miss  Cover   Missing
-----------------------------------------------------------------
aquarius/app/es_instance.py            90      4    96%   56-57, 61-62
aquarius/app/util.py                  137      5    96%   46-47, 92, 190, 194
aquarius/events/events_monitor.py     190     10    95%   157-158, 247, 294-301
aquarius/events/processors.py         171     10    94%   119, 182-186, 212, 267-272, 305-306
aquarius/events/purgatory.py           83     10    88%   109-112, 123-128
aquarius/events/request.py             22      1    95%   17
aquarius/events/util.py                94      5    95%   58, 65-67, 143
aquarius/myapp.py                      11      2    82%   21-22
aquarius/run.py                        54      6    89%   98-100, 104-110
-----------------------------------------------------------------
TOTAL                                1270     53    96%
```

**Pull request merged**